pkg/cmd/pulumi/convert.go

```diff
@@ -208,6 +207,14 @@ func runConvert(
 	}
 	defer contract.IgnoreClose(pCtx.Host)
 
+	// Translate well known sources to plugins
+	switch from {
+	case "tf":
+		from = "terraform"
+	case "":
+		from = "yaml"
```

Comment:
Would be nice to detect what we are converting from the files of the directory:
 - `Pulumi.yaml` exists, then it's `yaml`
 - `main.tf` exists (or any .tf files), then `terraform`
 - `main.pp` exists (or any .pp files) then `pcl`

Though this can be added later

Reply:
Yeh I've been pondering that idea. My first thought was to add a method to the converter plugins for the engine to ask if they could convert it. But firstly that's potentially slow if lots of converter plugins have to be started up and queried, but also relies on the plugins being installed which is a bit grim.
We could also have that logic in the engine here but that's means some converters get prefential treatment against others, plus tight coupling of the engine to all the converter types, both of which sounded lame.

So I'm currently assuming we won't do any automatic converter selection, but maybe a good  idea for it will come up later.